### ISSUE TITLE
Better error message in case of HTTP request failure

### ISSUE THREAD

**Issue by szpak** [Codearte/gradle-nexus-staging-plugin#5]

**Comment by marcingrzejszczak** [comment]
Definite +1 :)

```
spock-subjects-collaborators-extension (master ⚡→=) ./gradlew closeRepository promoteRepository
Starting a new Gradle Daemon for this build (subsequent builds will be faster).
:closeRepository
POST response data: null
:promoteRepository FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':promoteRepository'.
> Unexpected state of reveived repository. Received open, expected closed

```

it turned out that I have forgotten to upload my public key but I had to access Nexus to check this out.

also this 

```
POST response data: null
```

could be changed to sth like this

```
POST sent successfully
```

or sth like that - it might seem confusing for some people.

**Comment by szpak** [comment]
Thanks for your comment. I hope that REST response from Nexus contains that information (not only plain status 400...).

**Comment by szpak** [comment]
Btw, in the meaning in case of similar issues as a workaround you could use a `--debug` switch to display a complete HTTP communication with the server (it is very verbose, so redirecting to a file is recommended).

**Comment by szpak** [comment]
In a case of 500 from the server (with HTML content) a message is also not very meaningful:

```
:myProject:promoteRepository
Error parsing 'text/html' response
groovy.json.JsonException: Unable to determine the current character, it is not a string, number, array, or object

The current character read is '<' with an int value of 60
Unable to determine the current character, it is not a string, number, array, or object
line number 1
index number 0
<html>
^
```

More details #15.

**Comment by deanhiller** [comment]
fixed in pull request https://github.com/Codearte/gradle-nexus-staging-plugin/pull/32

@marcingrzejszczak 

I print the entire body, status code and status string and sonatype had some great error message in the body.

**Comment by marcingrzejszczak** [comment]
nice! :)

**Comment by javerstravis** [comment]
Hi, guys, I'm trying to move my deployment process from bintray back to sonatype
but this error occurs:

```
Execution failed for task ':promoteRepository'.
> Unexpected state of reveived repository. Received open, expected closed
```
Sure that error message could be better but what is the roor cause, what I'm doing wrong?

https://travis-ci.org/javers/javers/builds/207844456

**Comment by bartoszwalacik** [comment]
Hi, guys, I'm trying to move my deployment process from bintray back to sonatype
but this error occurs:

```
Execution failed for task ':promoteRepository'.
> Unexpected state of reveived repository. Received open, expected closed
```

Sure that error message could be better but what is the roor cause, what I'm doing wrong?

https://travis-ci.org/javers/javers/builds/207844456

**Comment by bartoszwalacik** [comment]
ok, got it. I've copied this config from the hermes project and it helped. Looks like you need to adjust the defaults

```
nexusStaging {
    numberOfRetries = 10
    delayBetweenRetriesInMillis = 3000
}
```

**Comment by szpak** [comment]
@bartoszwalacik Have you read [FAQ](https://github.com/Codearte/gradle-nexus-staging-plugin#faq)? :)

(Update: the problem described in FAQ is a little bit different - if you think it would be good to add your explicitly it can be done)

**Comment by szpak** [comment]
Btw, in a case of further problems running with `--info` can be useful to determine the source of your problem (by yourself). There is a plan to make it more clear even without enabling info logging - #34.

**Comment by szpak** [comment]
And yes, it will be adjusted: #12 (most likely in Q2 2017 just after 0.6.0 with rewritten backend will be finally released).

**Comment by szpak** [comment]
Implemented in #32. Available in 0.6.0.
